sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -0,0 +1,792 @@
+## API Report File for "@azure/ai-text-analytics"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AzureKeyCredential } from '@azure/core-auth';
+import { CommonClientOptions } from '@azure/core-client';
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface ActionCommon {
+    disableServiceLogs?: boolean;
+}
+
+// @public
+export type ActionPrebuilt = ActionCommon & {
+    modelVersion?: string;
+};
+
+// @public
+export const AnalyzeActionName: readonly ["EntityLinking", "EntityRecognition", "KeyPhraseExtraction", "PiiEntityRecognition", "LanguageDetection", "SentimentAnalysis"];
+
+// @public
+export type AnalyzeActionName = typeof AnalyzeActionName[number];
+
+// @public
+export type AnalyzeActionParameters<ActionName extends AnalyzeActionName> = ActionName extends "EntityLinking" ? EntityLinkingAction : ActionName extends "EntityRecognition" ? EntityRecognitionAction : ActionName extends "KeyPhraseExtraction" ? KeyPhraseExtractionAction : ActionName extends "PiiEntityRecognition" ? PiiEntityRecognitionAction : ActionName extends "SentimentAnalysis" ? SentimentAnalysisAction : ActionName extends "LanguageDetection" ? LanguageDetectionAction : never;
```

Comment:
rather than chaining `extends` I think we can index into an interface that maps the action types to options / results like how we did this in EventGrid: https://github.com/Azure/azure-sdk-for-js/blob/7a79e9ffd0d8d9c42ecec19d307faa0a8fa4aa5f/sdk/eventgrid/eventgrid/src/eventGridClient.ts#L49

Reply:
Done!